recipe/meta.yaml

```
@@ -0,0 +1,58 @@
+{% set name = "trulens-core" %}
+{% set version = "1.0.7" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/trulens_core-{{ version }}.tar.gz
+  sha256: bfbadd53a5a5d17575a8855f3fd888cad47912eb53fc4b0f27487ad341ba9971
+
+build:
+  script: {{ PYTHON }} -m pip install . -vv --no-deps --no-build-isolation
+  number: 100
+  skip: true  # [py<39]
+
+requirements:
+  host:
+    - python
+    - poetry-core
+    - pip
+  run:
+    - python
+    - numpy >=1.23
+    - munch >=2.5.0,<3.0.0
+    - dill >=0.3.8
+    - typing_extensions >=4.9.0,<5.0.0
+    - pydantic >=2.0.0,<3.0.0
+    - pandas >=2.1.0,<3.0.0
+    - rich >=13.6.0,<14.0.0
+    - sqlalchemy >=2.0.0,<3.0.0
+    - alembic >=1.8.1,<2.0.0
+    - nest-asyncio >=1.5.0,<2.0.0
+    - python-dotenv >=0.21,<2.0
```

```suggestion
    - python-dotenv >=0.21,<2.0
  run_constrained:
    - tqdm >=4.2.0
```
Not a blocker, but we could add tqdm.

---

Out of curiosity, how did you find `tqdm` to be a run-constrained?

---

It's listed as optional [here](https://github.com/truera/trulens/blob/trulens-1.1.0/src/core/pyproject.toml#L47). So, if a user decides they want to use it, the restriction should apply.